### PR TITLE
New logging paths

### PR DESCRIPTION
**Pull request by twolodzko**
- New CI configuration as in Prophet integration repo.
- New e2e test.
- Logging paths as in spec.
- Saving model summary on training start.

### CODE REVIEW

File: neptune_tensorflow_keras/impl/__init__.py [hunk 1]
```diff
@@ -116,24 +119,33 @@ def _log_metrics(self, logs, category: str, trigger: str):
         if not logs:
             return
 
-        logger = self._metric_logger[category][trigger]
+        logger = self._metric_logger["training"][category][trigger]
```
Comment (kshitij12345) [inline neptune_tensorflow_keras/impl/__init__.py:122]: Shouldn't this be `self._metric_logger[category]` as `base_namespace` should take care of adding `training` or anything that user specified?

Comment (twolodzko) [reply neptune_tensorflow_keras/impl/__init__.py:122]: @kshitij12345 it's a question for you. The spec used "training" in paths, I assumed we want this to be a fixed name?

Comment (kshitij12345) [reply neptune_tensorflow_keras/impl/__init__.py:122]: This discussion is resolved offline :)

Comment (twolodzko) [reply neptune_tensorflow_keras/impl/__init__.py]: Fixed as we discussed.

File: neptune_tensorflow_keras/impl/__init__.py [hunk 2]
```diff
@@ -130,7 +130,7 @@ def _log_metrics(self, logs, category: str, trigger: str):
                 pass
 
     def on_train_begin(self, logs=None):  # pylint:disable=unused-argument
-        self._metric_logger["training/model/summary"] = _model_summary_file(self.model)
```
Comment (kshitij12345) [inline neptune_tensorflow_keras/impl/__init__.py]: Should we update `self._metric_logger` to `self.base_handler` or something else? As we use it log model summary and metrics (and also model checkpoint in future).

Comment (twolodzko) [reply neptune_tensorflow_keras/impl/__init__.py]: Why?

File: neptune_tensorflow_keras/impl/__init__.py [hunk 3]
```diff
@@ -92,23 +95,19 @@ class NeptuneCallback(Callback):
         You need to have Keras or Tensorflow 2 installed on your computer to use this module.
     """
 
-    def __init__(self, run: Run, base_namespace: Optional[str] = None):
+    def __init__(self, run: Run, base_namespace: str = "training"):
```
Comment (kshitij12345) [inline neptune_tensorflow_keras/impl/__init__.py:98]: I think we should mention in the PR description that there are BC-Breaking changes.

File: neptune_tensorflow_keras/impl/__init__.py (hunk 2 above)
Comment (kshitij12345) [reply neptune_tensorflow_keras/impl/__init__.py]: I meant the name, sorry. (as it is used for more than metric logging like logging model summary). 

I see it as a handler to base namespace so maybe `self.base_namespace_handler`.

File: neptune_tensorflow_keras/impl/__init__.py (hunk 3 above)
Comment (aniezurawski) [reply neptune_tensorflow_keras/impl/__init__.py:98]: Why do we even change it?

File: neptune_tensorflow_keras/impl/__init__.py [hunk 4]
```diff
@@ -120,20 +119,29 @@ def _log_metrics(self, logs, category: str, trigger: str):
...
+    def on_test_end(self, logs=None):  # pylint:disable=unused-argument
+        self._log_metrics(logs, "test", "epoch")
+
 
-    def on_test_end(self, logs=None):
-        self._log_metrics(logs, 'test', 'epoch')
+def _model_summary_file(model) -> File:
+    stream = io.StringIO()
+    model.summary(print_fn=lambda x: stream.write(x + "\n"))
+    return File.from_stream(stream, extension="txt")
```
Comment (aniezurawski) [inline neptune_tensorflow_keras/impl/__init__.py:147]: Are you sure txt is good extension for model?

File: neptune_tensorflow_keras/impl/__init__.py (hunk 3 above)
Comment (twolodzko) [reply neptune_tensorflow_keras/impl/__init__.py:98]: @aniezurawski that's what's in the spec.

File: neptune_tensorflow_keras/impl/__init__.py (hunk 4 above)
Comment (twolodzko) [reply neptune_tensorflow_keras/impl/__init__.py:147]: It's the result of Python's `print`, it wouldn't need an extension, but with txt we get a preview.

File: neptune_tensorflow_keras/impl/__init__.py (hunk 2 above)
Comment (twolodzko) [reply neptune_tensorflow_keras/impl/__init__.py]: Right now it's used only once, I'll stick with the rule of thumb that it should be refactored if it repeats at least three times.

File: neptune_tensorflow_keras/impl/__init__.py (hunk 4 above)
Comment (aniezurawski) [reply neptune_tensorflow_keras/impl/__init__.py:147]: Isn't it a json/xml or some other well known format that we can get preview of?

File: neptune_tensorflow_keras/impl/__init__.py (hunk 3 above)
Comment (aniezurawski) [reply neptune_tensorflow_keras/impl/__init__.py:98]: > I think we should mention in the PR description that there are BC-Breaking changes.

And add some info to CHANGELOG.

File: neptune_tensorflow_keras/impl/__init__.py (hunk 4 above)
Comment (twolodzko) [reply neptune_tensorflow_keras/impl/__init__.py:147]: @aniezurawski nope, it's something like this:

```
Model: "sequential"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
=================================================================
 flatten (Flatten)           (None, 784)               0         
                                                                 
 dense (Dense)               (None, 10)                7850      
                                                                 
=================================================================
Total params: 7,850
Trainable params: 7,850
Non-trainable params: 0
_________________________________________________________________
```

Comment (aniezurawski) [reply neptune_tensorflow_keras/impl/__init__.py:147]: Hah. I imagine trying to parse such thing is nightmare. :P
Ok, fine. :)

File: neptune_tensorflow_keras/impl/__init__.py (hunk 3 above)
Comment (twolodzko) [reply neptune_tensorflow_keras/impl/__init__.py:98]: Done.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,12 @@
+## neptune-tensorflow-keras 1.0.0
+
+### Changes
+
+- `NeptuneCallback` now logs everything according to the `base_namespace` argument that defaults to `training` for
```
Comment (aniezurawski) [inline CHANGELOG.md:5]: Breaking change - passing None explicitly is not accepted anymore.

**Review by aniezurawski — APPROVED** [review]

**Pull request merged**